--- a/patch.sh
+++ b/patch.sh
@@ -234,8 +234,8 @@
         echo "WARNING: skipping extract of ${1} as it seems already extracted in ${TDIR}"
         return 0
     fi
-    [ -e ${1} ] || die "package ${1} not found"
-    sh ${1} --extract-only --target ${TDIR}
+    [ -e $BASEDIR/${1} ] || die "package ${1} not found"
+    sh $BASEDIR/${1} --extract-only --target ${TDIR}
     echo >> ${TDIR}/kernel/nvidia/nvidia.Kbuild
     chmod -R u+w ${TDIR}
 }
@@ -367,7 +367,7 @@
 
 if $DO_WSYS && [ ! -e "${WSYS}/nvlddmkm.sys" ]; then
     echo "source ${WSYS}/nvlddmkm.sys not found, will try to extract the installer"
-    NV_WIN_DRV_INSTALLER=`ls -1 ${VER_TARGET}*[-_]win*[-_]64bit*.exe | head -n 1`
+    NV_WIN_DRV_INSTALLER=`ls -1 ${BASEDIR}/${VER_TARGET}*[-_]win*[-_]64bit*.exe | head -n 1`
     [ -e "$NV_WIN_DRV_INSTALLER" ] || die "nvidia windows driver installer version $VER_TAGET not found"
     which 7z &>/dev/null || die "install p7zip-full for 7z tool (http://p7zip.sourceforge.net/)"
     which msexpand &>/dev/null || die "install mscompress (https://github.com/stapelberg/mscompress)"
